### DISCUSSION TITLE
Error en la compilacion

### DISCUSSION THREAD

**Discussion by salmantino** [tttapa/Control-Surface#747] [Q&A]
Buenas tardes, al compilar un codigo, me sale el siguiente error
In file included from f:\Documentos WINDOWS7\Mis documentos\Arduino\libraries\Control-Surface-master\src/AH/Hardware/LEDs/MAX7219_Base.hpp:11:0,
                 from f:\Documentos WINDOWS7\Mis documentos\Arduino\libraries\Control-Surface-master\src/AH/Hardware/LEDs/MAX7219SevenSegmentDisplay.hpp:6,
                 from f:\Documentos WINDOWS7\Mis documentos\Arduino\libraries\Control-Surface-master\src/Control_Surface.h:19,
                 from F:\Documentos WINDOWS7\Mis documentos\Arduino\MIDI_DLB_10\MIDI_DLB_10.ino:4:
C:\Users\Miguel\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.8.5\libraries\SPI\src/SPI.h:309:38: error: macro "attachInterrupt" requires 3 arguments, but only 1 given
   inline static void attachInterrupt() { SPCR |= _BV(SPIE); }
                                      ^
A que es debido y cual es la solucion

Un saludo ....... Miguel

**Comment by tttapa** [comment]
I'm unable to reproduce this issue. Neither the Arduino Core nor Control Surface define `attachInterrupt` as a macro, so I suspect this happens either in your sketch or in one of the other libraries you're using.

**Comment by salmantino** [comment]
Gracias por la contestacion, adjunto el programa, va con extension txt
[MIDI_DLB_10.txt](https://github.com/tttapa/Control-Surface/files/8748585/MIDI_DLB_10.txt)

**Comment by tttapa** [comment]
This is the culprit:

https://github.com/PaulStoffregen/Encoder/blob/8a3ebcae9f9e99cddb562348809c4be76c7a8c2b/utility/interrupt_config.h#L6

Please upgrade your version of Control Surface so you don't need the Encoder library, and remove the `#include <Encoder.h>` from your sketch.

**Comment by salmantino** [comment]
Muchas gracias por su respuesta, lo probe y solucionado

Un saludo ....... Miguel
